kamu-adapter-http/src/messages/smart_transfer_protocol_messages.rs

```diff
@@ -0,0 +1,198 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use url::Url;
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use opendatafabric::Multihash;
+
+
+/// Initial dataset pull request message
+#[derive(Serialize, Deserialize, Clone, PartialEq, Eq, Debug)]
+pub struct DatasetPullRequest {
+    pub begin_after: Option<Multihash>,
+    pub stop_at: Option<Multihash>,
+}
+
+
```

Comment:
Right, somehow it was disabled originally since Day 1 of my work.
I've ran formatting all over the project.
Btw, maybe we should think of installing it as a pre-commit hook.